### PR TITLE
Intern-5

### PR DESCRIPTION
**Pull request by olegstasiw**
- Bind data by Combine

### CODE REVIEW

File: Countries Info/ViewControllers/CountriesListViewController.swift [hunk 1]
```diff
@@ -6,30 +6,30 @@
...
+    var subscription = Set<AnyCancellable>()
+
     override func viewDidLoad() {
         super.viewDidLoad()
         appointDelegates()
         registerCell()
-   
         setUI()
+        
+        viewModel?.countriesPublisher.sink { _ in
```
Comment (romanmiller) [inline Countries Info/ViewControllers/CountriesListViewController.swift]: put this into some private func like -> configureObservers or other naming.

File: Countries Info/ViewModels/CountriesListViewModel.swift [hunk 2]
```diff
@@ -6,31 +6,41 @@
 //
 
 import Foundation
+import Combine
 
 protocol CountriesListViewModelProtocol {
     var countries: [Country]? { get }
```
Comment (romanmiller) [inline Countries Info/ViewModels/CountriesListViewModel.swift]: remove this from protocol, as out of ViewModel we should know just for Publisher.

**Review by romanmiller — APPROVED** [review]
Looks good to me.

File: Countries Info/ViewModels/CountriesListViewModel.swift [hunk 3]
```diff
@@ -6,31 +6,40 @@
...
     func numberOfItem() -> Int
     func cellViewModel(indexPath: IndexPath) -> CountryCollectionViewCellViewModelProtocol?
+    func fetch()
 }
 
-class CountriesListViewModel: CountriesListViewModelProtocol {
+class CountriesListViewModel: CountriesListViewModelProtocol, ObservableObject {
     
-    var countries: [Country]?
+    @Published var countries: [Country]?
```
Comment (trudakevych) [inline Countries Info/ViewModels/CountriesListViewModel.swift]: Do we need this array to be optional? 
Please check if we can use empty array as a default value, thanks

**Review by trudakevych — APPROVED** [review]

**Pull request merged**